localPlanner:
  # 常用参数
  dirThre: 75  # 目标点方向允许规划范围
  # 用于规划的点云的参数
  terrainVoxelSize: 0.05  # 地形分析的体素大小
  checkObstacle: true  # 控制是否根据用于的规划点云进行地形障碍检查
  checkRotObstacle: true  # 长条形车或差速底盘过窄道，限制转弯
  close_map_time: 2.5  # 关闭局部地图的超时时间
  use_fail_closemap: true  # 开启规划失败就关闭点云地图
  adjacentRange: 3.1  # 第一次裁剪规划点云的范围
  pathRangeStep: 0.35  # 上面这个系数的缩小步长
  minSpeedRange: 1.2
  pathRangeBySpeed: false  # 根据速度调整规划范围
  # 障碍物高度相关控制
  useCost: false  # 是否开启路径评分处理
  groundHeightThre: -0.1  # 点云高度大于这个阈值会惩罚得分
  costHeightThre: 0.1  # 高度得分最低不能低于这个值
  costScore: 0.02  # 高度得分的最低值
  pointPerPathThre: 1  # 同一块体素位置需要的障碍物点云数量
  # 目标点或转弯等方向信息的控制
  dirWeight: 80  # 评分中，只允许路径终点在这个转弯角度内的路径通过
  detourWei: 1.2  # 绕行系数
  pathCropByGoal: true  # 删除超过圆形区域的路径点
  goalClearRange_global: 10.0  # 全局范围
  # scale点云收拢比例控制
  usePathScale: false
  defPathScale: 1.0  # 默认缩放范围
  minPathScale: 1.0
  pathScaleStep: 0.2
  pathScaleBySpeed: false


pathFollower:
  # sports parameters
  yawRateGain: 2.0        # 一般情况下的车头转动增益
  stopYawRateGain: 3.0    # 静止下的车头转动增益
  maxYawRate: 35.0        # 车头最大转动角速度
  maxStopYawRate: 45.0    # 静止时车头最大转动角速度
  dirDiffThre_slow: 0.2   # 角度差
  dirDiffThre_keep: 0.05  # 角度差
  quick_turn_speed: 0.05  # 减速到这个速度就允许快速转向
  # get goal parameters
  close_direct_speed: 0.15    # 到目标点附近直冲的速度
  closeGoal_direct_dis: 0.6   # 到目标点附近直走的阈值&&达到getGoal_speed的距离
  goalSlowDisThre: 1.8        # 靠近全局目标点时减速
  getGoal_speed: 0.05         # 刚好追踪到目标点时的速度
  # Generally fixed parameters
  lookAheadDis: 0.5  # 每次只从车体当前位置往前lookAheadDis的距离的目标点来确定路径相对车头的方向
  useLoaclSlow: true  # 启用局部路径跟踪减速
  endPathDis: 0.05  # 跟踪到局部路径的末尾多少距离内就不再跟踪
  pathSlowDisThre: 0.4  # 局部路径跟踪开始减速的距离阈值, 局部路径最长就1m
  getPath_speed: 0.1  # 追踪到局部路径末端时的速度
  path_zero_bias: 0.01  # 追踪路径减速的零点偏移
  goal_zero_bias: 0.0  # 到点减速的零点偏移
  cloudSlow_minSpeed: 0.2  # 根据周围障碍物减速的最小速度
  curvature: 9  # 根据周围障碍物减速曲线的曲率，越小前面下降得越快，5到15左右就好
  slowBegin: 2  # 根据周围障碍物减速曲线的曲率，越小前面下降得越快，5到15左右就好
  goal_path_direct: true  # 是否全局路径直接追踪
  use_MIDPlanning_slow: true  # 是否使用转弯点减速
  MIDPlanning_slow_rate: 0.7  # 转弯点减速比例
  MIDPlanning_minSpeed: 1.0
  use_virtual_head: false
  use_move_base: false